### PR TITLE
#197 Add current options as part of selected options

### PR DESCRIPTION
**Pull request by Salam-Dalloul**

### CODE REVIEW

**Review by zsinnema — APPROVED** [review]

File: frontend/src/components/Forms/StatementForm.tsx [hunk 1]
```diff
@@ -1,419 +1,529 @@
...
-                    CustomFooter: CustomFooter,
-                },
-            },
+          label: "From",
+          placeholder: "Look for connections",
+          searchPlaceholder: "Search for connections",
+          noResultReason: "No prior connections found",
+          disabledReason: "",
+          onSearch: async (searchValue: string, formId: string) => {
+            return getAnatomicalEntitiesForForm(
```
Comment (afonsobspinto) [inline frontend/src/components/Forms/StatementForm.tsx:264]: This doesn't seem right. It should be using `searchFromEntitiesVia`

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
The search on the from is different from the search on the anatomical entities but it's being treated equally in the current PR. Please keep the logic from 'searchFromEntitiesVia' and 'searchFromEntitiesDestination'

File: frontend/src/services/CustomDropdownService.ts [hunk 2]
```diff
@@ -1,217 +1,288 @@
...
+  const viaIndex = getFirstNumberFromString(formId);
+  if (
+    viaIndex == null ||
+    statement.vias == null ||
+    statement.vias[viaIndex] == null
+  ) {
+    return [];
+  }
+  const viaOrder = statement.vias[viaIndex].order;
+  const anatomicalEntities = getEntitiesBeforeOrder(statement, viaOrder);
```
Comment (afonsobspinto) [inline frontend/src/services/CustomDropdownService.ts:128]: Although practically speaking it's unlikely that we will need it, we should still use the current selected options in the searchFromEntitiesX functions to populate the anatomicalEntities options (without creating duplicates)

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
Looks better but see my comment above :arrow_up:

**Review by afonsobspinto — APPROVED** [review]

**Comment by zsinnema** [comment]
@Salam-Dalloul can you have a look at the conflicts? Thanks!

**Comment by zsinnema** [comment]
@Salam-Dalloul still a conflict remains. Can you have a look? thanks!

**Pull request merged**